src/main/java/org/opensearch/security/ssl/SslContextHandler.java

```diff
@@ -68,40 +70,69 @@ Stream<Certificate> keyMaterialCertificates(final List<Certificate> certificates
         return certificates.stream().filter(Certificate::hasKey);
     }
 
+    Stream<Certificate> authorityCertificates() {
+        return authorityCertificates(loadedCertificates);
+    }
+
+    Stream<Certificate> authorityCertificates(final List<Certificate> certificates) {
+        return certificates.stream().filter(not(Certificate::hasKey));
+    }
+
     void reloadSslContext() throws CertificateException {
         final var newCertificates = sslConfiguration.certificates();
 
-        if (sameCertificates(newCertificates)) {
-            return;
+        boolean hasChanges = false;
+
+        final var loadedAuthorityCertificates = authorityCertificates().collect(Collectors.toList());
+        final var loadedKeyMaterialCertificates = keyMaterialCertificates().collect(Collectors.toList());
+        final var newAuthorityCertificates = authorityCertificates(newCertificates).collect(Collectors.toList());
+        final var newKeyMaterialCertificates = keyMaterialCertificates(newCertificates).collect(Collectors.toList());
+
+        if (notSameCertificates(loadedAuthorityCertificates, newAuthorityCertificates)) {
+            hasChanges = true;
+            validateDates(newAuthorityCertificates);
```

Comment:
Curious, why not always perform validateDates?

Will newAuthorityCerts always have a different signature then loadedAuthorityCerts?